client/apiclient.go

```diff
@@ -46,6 +48,18 @@ type TLSConfig struct {
 	Insecure bool
 }
 
+// IsDefault checks whether a TLSConfig structure has had any fields changed from their defaults.
+// If any field is non-default, it returns false; otherwise, it returns true.
+func (t TLSConfig) IsDefault() bool {
```

Comment:
As long as the struct doesn't need default state set in a `New{$StructName}` function  go you can check if `t` is equal to the struct type with no values set: e.g. `if t == TLSConfig{} {...}`.

For checking equality with nested collections (slice, map) and struct, `reflect.DeepEqual()` will do the job so long as all the values are truly equal rather than semantically. 

Here's an example, I believe we can use reflect.DeepEqual() in this case. https://go.dev/play/p/U9ycWaVh1mO

I've also been reading that "github.com/google/go-cmp/cmp"'s `cmp.Equal()` is useful for checking semantic equality (e.g. two different slices with the same values should be equal), but I have not used it yet!

Reply:
Thanks! I tried the `if t == TLSConfig{}` approach, but realized there is a `[]byte` in the struct, which prevents direct comparison. I hadn't considered `reflect.DeepEqual()` - I'll use that instead!

go-cmp sounds useful! For the equality check in this particular case, I think I'll stick with reflection, but it sounds like it could be handy to keep in mind for the future!